content/en/boilerplates/snips/args.sh

```diff
@@ -21,7 +21,7 @@
 ####################################################################################################
 
 ! read -r -d '' bpsnip_args_gateway_api_version <<\ENDSNIP
-v0.8.0
+f98e94a5305e3a08cb14f8a03470f8f3bdf6d54c
```

Comment:
I don't recall why we need to strip anything here. We really just want the docs to use the same version that the istio/istio is using, currently `v0.8.2-0.20230915021212-f98e94a5305e`

Reply:
@frankbu That's exactly what I hardcoded. That last part of `v0.8.2-0.20230915021212-f98e94a5305e` is the commit sha, and `f98e94a5305e3a08cb14f8a03470f8f3bdf6d54c` is the full sha. I do not currently have the makefile doing that conversion.

Reply:
If I try and strip the trailing items off (what the code used to do), you get `v0.8.2` and `github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.8.2` and since that release doesn't exist, the customize fails.

Reply:
Ah, right. So the problem is that when it's an official release we want to use it instead? Maybe the makefile can check if it's a proper release, use it, otherwise use the commit sha?

Reply:
A full blown solution might be if there are `-`'s, then we know we have to do some special logic. For RC's we strip of what's after the RC since that will exist). For others we have to strip of what's after the - (so here it would be `v0.8.2`) and then decrement the version (to v0.8.1). And we need special logic on case we have to go from v0.8.0 to 0.7.x (whatever the last one was (we can't just cement to 9).

Reply:
Yeah, but then we couldn't doc new things until after they're officially released. Seems that whenever there are `-` chars, the last part is a prefix of the sha we need. Isn't there a way to automatically find the full sha, given the prefix (i.e., `f98e94a5305e` -> `f98e94a5305e3a08cb14f8a03470f8f3bdf6d54c`)?

Reply:
The only way I know to get the full sha from the short sha is to git fetch the repo, git checkout the short sha and then get the full sha (we do this in bin/init.sh), but we are getting the repo so we can build istioctl for the appropriate commit. This seems excessive for here, but we could do it.

Reply:
You can do it without cloning the repo which might be faster using the GitHub API. https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#get-a-commit -- If the repo is public, you do not need to provide an Authorization key and `REF` in this case accepts both short and full SHAs. The returned sha will always be full. For example:

```
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/kubernetes-sigs/gateway-api/commits/f98e94a5305e | jq -r .sha
```

Will return `f98e94a5305e3a08cb14f8a03470f8f3bdf6d54c`

Reply:
Thanks Daniel, let me try that.